### PR TITLE
ci from scratch

### PR DESCRIPTION
**Pull request by cameron1024**

### CODE REVIEW

**Comment by cameron1024** [comment]
Caching is working, unfortunatey github actions is slow, takes 10m to run clippy even with a cache :(

File: flake.nix [hunk 1]
```diff
@@ -278,35 +218,27 @@
...
+            openssl
+            protobuf
+            uniffi-bindgen
+            postgresql
+            diesel-cli
+            cargo-insta # snapshot testing lib
+          ];
           shellHook = ''
             echo "=== Catalyst Core development shell ==="
+            echo "Info: Git hooks can be installed using \`pre-commit install\`"
```
Comment (saibatizoku) [inline flake.nix:237]: This is no longer used.

File: flake.nix [hunk 2]
```diff
@@ -278,35 +218,27 @@
...
           '';
-          # TODO: is this needed for vit-testing development
-          # export PATH="${jormungandr}/bin:$PATH"
-          # export PATH="${vit-servicing-station-server}/bin:$PATH"
         };
 
+        defaultPackage = warnToUpdateNix packages.default;
```
Comment (saibatizoku) [inline flake.nix:241]: Monorepo has no default package defined. This no longer applies.

**Review by saibatizoku — APPROVED** [review]
Looks good, thanks for the CI!

File: .github/workflows/ci_tests.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - uses: Swatinem/rust-cache@v2
+      - name: Install deps
+        run:
+          sudo apt install -y protobuf-compiler libssl-dev libpq-dev libsqlite3-dev pkg-config
+      - name: Clippy and fmt
```
Comment (cong-or) [inline .github/workflows/ci_tests.yml:66]: should we add _-- -D warnings_ for clippy somewhere?

Comment (cameron1024) [reply .github/workflows/ci_tests.yml:66]: I believe that's the default behaviour when running `cargo clippy` but I'll double check when I'm at my computer

**Review by FelipeRosa — APPROVED** [review]

File: .github/workflows/ci_tests.yml (hunk 3 above)
Comment (cameron1024) [reply .github/workflows/ci_tests.yml:66]: Yep, `cargo clippy` gives a non-zero exit code if any issues are found

**Pull request merged**